d3ledger/back-office
# Electron -> Web

We are getting rid of Electron since grpc-web is working: https://github.com/improbable-eng/grpc-web

Also, fixed a very strange bug which was caused by `util` alias :)

<img width="1552" alt="screen shot 2018-07-16 at 12 38 05" src="https://user-images.githubusercontent.com/10958990/42751957-2003e124-88f5-11e8-8cba-91e699468c9d.png">

## How to test
Start `grpcwebproxy`:
1. https://gitlab.com/snippets/1713665 download docker file and run `docker build -t grpcwebproxy .`
2. Start docker-container with
```
docker run -it --rm \
  --network="iroha-network" \
  -p 8080:8080 \
  -P grpcwebproxy \
  --backend_addr=iroha:50051 \
  --run_tls_server=false
```
--network="iroha-network" is required if you started iroha following http://iroha.readthedocs.io/en/latest/getting_started/
`--backend_addr` should point at Iroha instance.

Now start D3 with `yarn serve`, but enter ip address of a proxy (e.g. http://localhost:8080)

## Future considerations
1. Learn how https://github.com/improbable-eng/grpc-web works and look, if we could use typescript files instead of `iroha-lib`.
2. Remove proto files which are not needed + put those proto files in iroha-lib (proto-files are generated using https://github.com/improbable-eng/grpc-web/blob/master/ts/docs/code-generation.md)
